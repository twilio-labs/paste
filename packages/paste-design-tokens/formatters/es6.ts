import * as lodash from 'lodash';
import {ImmutableStyleMap} from 'theo';
import {Token} from '../types';
import {getTokenCategories} from '../utils/getTokenCategories';
import {formatSingleTokensWithTemplate} from '../utils/formatSingleTokensWithTemplate';
import {formatGroupTokensWithTemplate} from '../utils/formatGroupTokensWithTemplate';

export const tokenTemplate = ({name, value}: {name: string; value: string}): string =>
  `export const ${lodash.camelCase(name)} = "${value.replace(/"/g, '\\"')}";`;

export const categoryTemplate = (categoryName: string, props: Token[]): string => `export const ${lodash.camelCase(
  categoryName
)} = {
${props.map(prop => `${lodash.camelCase(prop.name)},`).join('\n')}
};`;

export const es6TokenFormat = (result: ImmutableStyleMap): string => {
  const singleTokens = formatSingleTokensWithTemplate(result, tokenTemplate);

  const categories = getTokenCategories(result);

  const groupedTokens = formatGroupTokensWithTemplate(result, categories, categoryTemplate);

  return [singleTokens, groupedTokens].join('\n');
};
