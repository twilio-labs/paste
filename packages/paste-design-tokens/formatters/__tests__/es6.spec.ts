import * as theo from 'theo';
import {resolve} from 'path';
import {es6TokenFormat} from '../es6';

theo.registerFormat('es6.js', es6TokenFormat);

describe('es6Formatter', () => {
  it('should return es6 formatted tokens', () => {
    return theo
      .convert({
        transform: {
          type: 'web',
          file: resolve(__dirname, '../__fixtures__/index.yml'),
        },
        format: {
          // @ts-ignore Theo isn't typed for custom format types
          type: 'es6.js',
        },
      })
      .then((es6JS: string) => {
        return expect(es6JS).toMatchSnapshot();
      })
      .catch((error: string) => {
        console.log(`Something  went wrong: ${error}`);
        throw new Error('[es6Formatter test]: should return es6 formatted tokens');
      });
  });
});
