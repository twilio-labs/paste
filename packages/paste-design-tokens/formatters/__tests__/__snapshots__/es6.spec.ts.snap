// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`es6Formatter should return es6 formatted tokens 1`] = `
"export const colorBackgroundBody = \\"rgb(255, 255, 255)\\";
export const colorBackgroundBrandHighlight = \\"rgb(242, 47, 70)\\";
export const colorBackgroundBrandHighlightDark = \\"rgb(235, 15, 41)\\";
export const colorBackgroundBrandHighlightDarker = \\"rgb(194, 37, 55)\\";
export const colorBackgroundBrandPrimary = \\"rgb(13, 18, 43)\\";
export const colorBackgroundDefault = \\"rgb(246, 246, 246)\\";
export const colorBackgroundElementSelection = \\"rgb(236, 187, 193)\\";
export const colorBackgroundError = \\"rgb(206, 80, 95)\\";
export const colorBackgroundErrorDark = \\"rgb(174, 33, 49)\\";
export const colorBackgroundErrorLight = \\"rgb(194, 37, 55)\\";
export const colorBackgroundRowActive = \\"rgb(230, 230, 230)\\";
export const colorBackgroundRowFocus = \\"rgb(214, 214, 214)\\";
export const colorBackgroundRowHover = \\"rgb(214, 214, 214)\\";
export const colorBackgroundRowSelected = \\"rgb(230, 230, 230)\\";
export const colorBackgroundRowStriped = \\"rgb(235, 235, 235)\\";
export const colorBackgroundSuccess = \\"rgb(100, 209, 138)\\";
export const colorBackgroundSuccessDark = \\"rgb(80, 167, 110)\\";
export const colorBackgroundSuccessLight = \\"rgb(162, 227, 184)\\";
export const colorBackgroundWarning = \\"rgb(251, 208, 144)\\";
export const colorBackgroundWarningDark = \\"rgb(199, 141, 56)\\";
export const colorBackgroundWarningLight = \\"rgb(249, 177, 71)\\";
export const colorBrandHighlight = \\"rgb(242, 47, 70)\\";
export const colorBrandHighlightDark = \\"rgb(235, 15, 41)\\";
export const colorBrandPrimary = \\"rgb(13, 18, 43)\\";
export const colorGray1 = \\"rgb(255, 255, 255)\\";
export const colorGray10 = \\"rgb(34, 34, 34)\\";
export const colorGray2 = \\"rgb(246, 246, 246)\\";
export const colorGray3 = \\"rgb(235, 235, 235)\\";
export const colorGray4 = \\"rgb(230, 230, 230)\\";
export const colorGray5 = \\"rgb(214, 214, 214)\\";
export const colorGray6 = \\"rgb(204, 204, 204)\\";
export const colorGray7 = \\"rgb(153, 153, 153)\\";
export const colorGray8 = \\"rgb(102, 102, 102)\\";
export const colorGray9 = \\"rgb(68, 68, 68)\\";
export const colorTextBrandHighlight = \\"rgb(242, 47, 70)\\";
export const colorTextBrandHighlightDark = \\"rgb(235, 15, 41)\\";
export const colorTextBrandPrimary = \\"rgb(255, 255, 255)\\";
export const colorTextDefault = \\"rgb(68, 68, 68)\\";
export const colorTextError = \\"rgb(194, 37, 55)\\";
export const colorTextIconDefault = \\"rgb(68, 68, 68)\\";
export const colorTextInverse = \\"rgb(255, 255, 255)\\";
export const colorTextInverseWeak = \\"rgb(235, 235, 235)\\";
export const colorTextLabel = \\"rgb(68, 68, 68)\\";
export const colorTextLink = \\"rgb(0, 112, 204)\\";
export const colorTextLinkActive = \\"rgb(0, 89, 163)\\";
export const colorTextLinkDestructive = \\"rgb(194, 37, 55)\\";
export const colorTextLinkDestructiveActive = \\"rgb(155, 29, 44)\\";
export const colorTextLinkDestructiveDisabled = \\"rgb(236, 187, 193)\\";
export const colorTextLinkDestructiveFocus = \\"rgb(174, 33, 49)\\";
export const colorTextLinkDestructiveHover = \\"rgb(174, 33, 49)\\";
export const colorTextLinkDisabled = \\"rgb(102, 169, 224)\\";
export const colorTextLinkFocus = \\"rgb(0, 97, 192)\\";
export const colorTextLinkHover = \\"rgb(0, 97, 192)\\";
export const colorTextPlaceholder = \\"rgb(102, 102, 102)\\";
export const colorTextSuccess = \\"rgb(100, 209, 138)\\";
export const colorTextWarning = \\"rgb(249, 177, 71)\\";
export const colorTextWeak = \\"rgb(230, 230, 230)\\";
export const fontSize0 = \\"0.6875rem\\";
export const fontSize1 = \\"0.75rem\\";
export const fontSize2 = \\"0.8125rem\\";
export const fontSize3 = \\"0.875rem\\";
export const fontSize4 = \\"0.9375rem\\";
export const fontSize5 = \\"1rem\\";
export const fontSize6 = \\"1.125rem\\";
export const fontSize7 = \\"1.25rem\\";
export const fontSize8 = \\"1.875rem\\";
export const fontSize9 = \\"2.25rem\\";
export const space0 = \\"0.25rem\\";
export const space1 = \\"0.5rem\\";
export const space10 = \\"2.75rem\\";
export const space11 = \\"3rem\\";
export const space12 = \\"3.25rem\\";
export const space13 = \\"3.5rem\\";
export const space14 = \\"3.75rem\\";
export const space15 = \\"4rem\\";
export const space2 = \\"0.75rem\\";
export const space3 = \\"1rem\\";
export const space4 = \\"1.25rem\\";
export const space5 = \\"1.5rem\\";
export const space6 = \\"1.75rem\\";
export const space7 = \\"2rem\\";
export const space8 = \\"2.25rem\\";
export const space9 = \\"2.5rem\\";
export const zIndex0 = \\"0\\";
export const zIndex10 = \\"10\\";
export const zIndex20 = \\"20\\";
export const zIndex30 = \\"30\\";
export const zIndex40 = \\"40\\";
export const zIndex50 = \\"50\\";
export const zIndex60 = \\"60\\";
export const zIndex70 = \\"70\\";
export const zIndex80 = \\"80\\";
export const zIndex90 = \\"90\\";
export const backgroundColors = {
colorBackgroundBody,
colorBackgroundBrandHighlight,
colorBackgroundBrandHighlightDark,
colorBackgroundBrandHighlightDarker,
colorBackgroundBrandPrimary,
colorBackgroundDefault,
colorBackgroundElementSelection,
colorBackgroundError,
colorBackgroundErrorDark,
colorBackgroundErrorLight,
colorBackgroundRowActive,
colorBackgroundRowFocus,
colorBackgroundRowHover,
colorBackgroundRowSelected,
colorBackgroundRowStriped,
colorBackgroundSuccess,
colorBackgroundSuccessDark,
colorBackgroundSuccessLight,
colorBackgroundWarning,
colorBackgroundWarningDark,
colorBackgroundWarningLight,
};
export const colors = {
colorBrandHighlight,
colorBrandHighlightDark,
colorBrandPrimary,
colorGray1,
colorGray10,
colorGray2,
colorGray3,
colorGray4,
colorGray5,
colorGray6,
colorGray7,
colorGray8,
colorGray9,
};
export const fontSizes = {
fontSize0,
fontSize1,
fontSize2,
fontSize3,
fontSize4,
fontSize5,
fontSize6,
fontSize7,
fontSize8,
fontSize9,
};
export const spacings = {
space0,
space1,
space10,
space11,
space12,
space13,
space14,
space15,
space2,
space3,
space4,
space5,
space6,
space7,
space8,
space9,
};
export const textColors = {
colorTextBrandHighlight,
colorTextBrandHighlightDark,
colorTextBrandPrimary,
colorTextDefault,
colorTextError,
colorTextIconDefault,
colorTextInverse,
colorTextInverseWeak,
colorTextLabel,
colorTextLink,
colorTextLinkActive,
colorTextLinkDestructive,
colorTextLinkDestructiveActive,
colorTextLinkDestructiveDisabled,
colorTextLinkDestructiveFocus,
colorTextLinkDestructiveHover,
colorTextLinkDisabled,
colorTextLinkFocus,
colorTextLinkHover,
colorTextPlaceholder,
colorTextSuccess,
colorTextWarning,
colorTextWeak,
};
export const zIndices = {
zIndex0,
zIndex10,
zIndex20,
zIndex30,
zIndex40,
zIndex50,
zIndex60,
zIndex70,
zIndex80,
zIndex90,
};"
`;
